### ISSUE TITLE
How to style Gutenberg components plugin-wide

### ISSUE THREAD

**Issue by tomalec** [woocommerce/google-listings-and-ads#747]
_This is a follow-up from issues discussed in https://github.com/woocommerce/google-listings-and-ads/pull/720#discussion_r644940450 and https://github.com/woocommerce/google-listings-and-ads/pull/732#issuecomment-854618373_

## User story
As a developer, I'd like to tweak styles for all instances of a specific Gutenberg or WooCommerce-admin component, without too much effort.

## Technical
Currently, we use two techniques and both have some flaws:
1. Wrap with a plugin-specific component. Like we do for https://github.com/woocommerce/google-listings-and-ads/blob/c2a223dd7aedcd0f9444d89657529b63a9b156b7/js/src/components/app-radio-content-control/index.js#L24 
	We create a new component locally, attach a local class name to it, then apply some CSS rules to this wrapper or its descendants
	- It requires some manual effort to wrap such component and keep the API up-to date
	- It requires us to remember to use it and not the imported component
2. We provide a stylesheet that hacks all Gutenberg components https://github.com/woocommerce/google-listings-and-ads/blob/6a2087c2fe353bc744d1c8cacf8922b1217eb256/js/src/css/shared/_gutenberg-components.scss
	- As we do not use style scoping, this stylesheet is applied to the entire document. If another plugin's component appears on the same page, we may unintentionally break its look and feel.

## Acceptance criteria
* [ ] The DevX ergonomic for a developer using an imported component should not be much worse that using non tweaked one. 
* [ ] Preferably, the tweak should be "applied" in one place, and users of a tweaked component should not have to even think about it, or at least mark it in a minimal manner.
* [ ] We should not break the other plugins.

## Out of bounds/rabbit holes

- As we are talking about tweaking/hacking Gutenberg components, each "tweak" would always be fragile to their component's changes, so the changes should be rather minimal,
- … or based on external component's documented behavior
- … and preferably joined with a PR with an enhancement to an external component. With a 
	- completely changed behavior
	- additional variant option
	- or exposed structure change. That would change intrusive hack, to a regular HTML composition.

## Event tracking
n/a

**Comment by eason9487** [comment]
> 2. We provide a stylesheet that hacks all Gutenberg components https://github.com/woocommerce/google-listings-and-ads/blob/6a2087c2fe353bc744d1c8cacf8922b1217eb256/js/src/css/shared/_gutenberg-components.scss
>     * As we do not use style scoping, this stylesheet is applied to the entire document. If another plugin's component appears on the same page, we may unintentionally break its look and feel.

Agree. I just realized that the js and css files of this plugin would be loaded when viewing other pages under the "Marketing" menu. I didn't expect this mechanism, and I'm not sure if this is the expected loading behavior of WP/WC plugins. In any case, it's actually necessary to make sure that our style adjustments for Gutenberg components do not go beyond this plugin scope.
